core/src/main/java/com/graphhopper/routing/ev/Country.java

```diff
@@ -279,7 +279,11 @@ public static EnumEncodedValue<Country> create() {
         for (Country c : Country.values()) map.put(c.getTwoLetterCode(), c);
     }
 
+    /**
+     * This method returns a Country enum for the specified ISO3166-1:alpha2 code.
+     */
     public static Country valueOfAlpha2(String alpha2) {
-        return alpha2 == null ? null : map.get(alpha2);
+        if (alpha2 == null || alpha2.length() != 2) return null;
+        return map.get(alpha2);
```

Comment:
I don't see why we shouldn't just add them. I would craft a PR if you don't have any objections.

Reply:
I am aware of that. But it might be good to offer alpha3 codes for those who integrate GraphHopper as a library. Otherwise they need additional dependencies like [nv-i18n](https://github.com/TakahikoKawasaki/nv-i18n/blob/master/src/main/java/com/neovisionaries/i18n/CountryCode.java) to convert back and forth from two- to three-letter codes.

Reply:
My usecase would rather be something like "Which countries is route XY passing?". If I need three letter codes an additional library is required to translate the two-letter code.

Reply:
Nevermind. I'm blind. We already support both of them.